src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -665,24 +666,40 @@ def update_vnet(instance, address_prefixes=None):
         instance.address_space.address_prefixes = address_prefixes
     return instance
 
+def _set_route_table(ncf, resource_group_name, route_table, subnet):
+    if route_table:
+        is_id = is_valid_resource_id(route_table)
+        rt = None
+        if is_id:
+            resource_id = parse_resource_id(route_table)
+            rt = ncf.route_tables.get(resource_id['resource_group'], resource_id['name'])
+        else:
+            rt = ncf.route_tables.get(resource_group_name, route_table)
+        subnet.route_table = rt
+    elif route_table == '':
+        subnet.route_table = None
+
 def create_subnet(resource_group_name, virtual_network_name, subnet_name,
-                  address_prefix='10.0.0.0/24', network_security_group=None):
+                  address_prefix='10.0.0.0/24', network_security_group=None,
+                  route_table=None):
     '''Create a virtual network (VNet) subnet
     :param str address_prefix: address prefix in CIDR format.
-    :param str network_security_group: attach with existing network security group,
-        both name or id are accepted.
+    :param str network_security_group: Name or ID of network security
+        group to assiciate with the subnet.
     '''
     ncf = _network_client_factory()
     subnet = Subnet(name=subnet_name, address_prefix=address_prefix)
     subnet.address_prefix = address_prefix
 
     if network_security_group:
         subnet.network_security_group = NetworkSecurityGroup(network_security_group)
+    _set_route_table(ncf, resource_group_name, route_table, subnet)
 
     return ncf.subnets.create_or_update(resource_group_name, virtual_network_name,
                                         subnet_name, subnet)
 
-def update_subnet(instance, address_prefix=None, network_security_group=None):
+def update_subnet(instance, args, address_prefix=None, network_security_group=None, #pylint: disable=unused-argument
```

Comment:
>args [](start = 28, length = 4)

This should just work with resource_group_name as a signature argument. Using dictionary-like access through generic args is one of the anti-patterns we sought to get rid of by switching to the "simple method" way of writing custom commands.